OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -300,6 +300,7 @@ public interface IMove
 		Activity MoveToTarget(Actor self, Target target);
 		Activity MoveIntoTarget(Actor self, Target target);
 		Activity VisualMove(Actor self, WPos fromPos, WPos toPos);
+		Activity MoveToDock(Actor host, Actor client, IDock dock, Activity requester, bool goThroughHost = false);
```

Comment:
I don't want to expose `IDock` to the core engine code, so I filed #13991 to fix this in a cleaner way.  After that has been merged you can rebase this and then reference the `Dock` trait directly here.